presto-main/src/main/java/com/facebook/presto/util/HashCollisionsEstimator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Math.ceil;
+import static java.lang.Math.floor;
+
+public final class HashCollisionsEstimator
```

Comment:
Those estimates scale accurately to larger hash tables. For instance assume you have estimated number of collisions `c` for inserting `x` entries into `y` sized hash table. Then the number of collisions for inserting `x*scale` entries into `y*scale` sized hash table is exactly `c*scale`. You can estimate the number of collisions in such way for hash table of any size. Your relative error will remain the same independently of table size. The greater number of estimates `N`, the better the estimate. However, I have found that with `500` data points I get really accurate (less than `1%` error) (and useful) results in explain analyze output. 

It's much easier to deduce closed formula for Birthday problem and hash tables where after collision you uniformly try next pos (instead of trying the next table cell linearly). In fact, I deduced formula for the latter one.

It is not possible though to deduce closed formula for hash tables where after collision you try the next table cell linearly. This is because you cannot assume uniform distribution of table entries when inserting new element (each recursion step is different). We though about closed formula with my mathematician colleague, but concluded that it is rather not possible (did not hard prove the fact through).

Therefore I propose we keep it estimated since it is a really nice feature (e.g: we've seem performance improvements due to hash computation optimizations by @martint) and allows user to identify potential data or hashing problems.